### DISCUSSION TITLE
Decorators with private class fields?

### DISCUSSION THREAD

**Discussion by joezappie** [babel/babel#14845] [Q&A]
I'm trying to switch a bunch of my web components to use Lit Element. I'd like to use Lit's decorators, but my classes already use a bunch of private fields. That's causing the error "Private fields in decorated classes are not supported yet" when using version "2018-09".

Switching to legacy version, I no longer get the private field error but Lit's documentation say it only supports version "2018-09" and it causes some other off-topic issues with the public field decorators and their getter/setters.

So it seems you either need to pick to use decorators or private fields. Is there a way to configure babel to convert any private # functions or properties to be public? This would allow coding to support the future spec even if its not supported now. 

I see a couple old issues about this, notable #9973 from 2019. It was stated it would eventually be supported, curious 3 years later if theres been any progress on that front?

**Comment by joezappie** [comment]
Actually, looking at the relative support for private fields/methods its now 90% which is good enough for me. Seems like I should just be able to disable the private plugins? I'll try this out tomorrow:

```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "exclude": [
           "@babel/plugin-proposal-private-methods",
            "@babel/plugin-proposal-private-property-in-object"
         ]
      }
    ]
  ]
}
```

**Comment by JLHwung** [comment]
You can try out the `"2021-12"` version and see if that works with Lit.

For future readers: If you are using Babel 7.19 or above, please update to the `"2022-03"` version.
